analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -55,58 +57,122 @@ private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberC
         else if (left.IsPositive)
         {
+            // At this point, right range R can be:
+            // - negative => it can contain -1 or other negative numbers, which can at most make the result equal to max of L.
+            // - mixed => it's the union of a negative sub-range Rn and a positive one Rp.
+            //   - L & Rn can go as high as max of L, because Rn contains -1 (all ones).
+            //   - L & Rp, because both are positive, cannot be bigger than the smaller of max of L and Rp, so smaller of max of L and R.
+            // Therefore, L & R <= max(max of L, smaller of max of L and R) = max of L.
             return left.Max;
         }
         else if (right.IsPositive)
         {
+            // Same reasoning as above
             return right.Max;
         }
         else
         {
-            // If both operands can be -1 and both can have positive values => The result cannot be bigger than the bigger of the two maxima.
-            // If one operand is negative and one can have positive values => The result cannot be bigger than the positive maximum.
+            // If both ranges are mixed => They both contain -1 => The result can be as big as than the bigger of the two maxima.
+            // If one range is negative and one is mixed => The result can be as big as the max of the mixed range => bigger of the two maxima.
             return BiggestMaximum(left, right);
         }
     }
 
+    private static BigInteger? CalculateOrMin(NumberConstraint left, NumberConstraint right)
+    {
+        // BitOr can only turn 0s into 1s, not the other way around => If both operands have the same sign, the result cannot be smaller than the bigger of the two.
+        if ((left.IsNegative && right.IsNegative) || (left.IsPositive && right.IsPositive))
+        {
+            return BiggestMinimum(left, right);
+        }
+        else if (left.IsNegative)
+        {
+            // At this point, right range R can be:
+            // - positive => it can contain 0 or other positive numbers, which can at most make the result as small as min of L.
+            // - mixed => it's the union of a negative sub-range Rn and a positive one Rp.
+            //   - L & Rn, because both are negative, cannot be smaller than the bigger of min of L and Rn, so bigger of min of L and R.
+            //   - L & Rp can go as small as min of L, because Rp contains 0 (all zeros).
+            // Therefore, L & R >= min(bigger of min of L and R, min of L) = min of L.
+            return left.Min;
+        }
+        else if (right.IsNegative)
+        {
+            // Same reasoning as above
+            return right.Min;
+        }
+        else
+        {
+            // If both ranges are mixed => They both contain 0 => The result can be as small as the smaller of the two minima.
+            // If one range is negative and one is mixed => The result can be as small as the min of the mixed range => smaller of the two minima.
```

Comment:
Can be simplified in the same way as And.

Reply:
Changed into `// If both ranges are mixed or one negative and one mixed => smaller of the two minima is just the general lower boundary for min.`.